src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -233,6 +243,11 @@ export default class GooglePayPaymentStrategy implements PaymentStrategy {
             const paymentData = await this._googlePayPaymentProcessor.displayWallet();
             await this._paymentInstrumentSelected(paymentData, methodId);
 
+            if (this._paymentMethod?.initializationData.nonce) {
+                const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId));
+                this._paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+            }
+
```

Comment:
Trying to fetch a new nonce should not be a responsibility of `_displayWallet`. The `_displayWallet` method knows nothing about the nonce nor uses it. 

After a quick scan of the code, it looks like the logic that refetches the nonce should either go in `_getPayment` or `_getNonce`, which are the 2 methods that know and care about the nonce.

Reply:
I add a new expect when the  when we have this error the method `getPaymentMethodOrThrow` should be called 3 times, this assert just happen when the payment method fails and we need to show the wallet again, let me know if this assertion sounds more sense for you or we need to test some else @icatalina 🙏🏼